File: src/main/java/ru/bebriki/bebriki/models/Worker.java
package ru.bebriki.bebriki.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
@Table(name = "workers")
public class Worker {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(name = "name")
    private String name;
    @Column(name = "surname")
    private String surname;
    @Column(name = "secondName")
    private String secondName;
    @Column(name = "login")
    private String login;
    @Column(name = "vacancyId")
    private Integer vacancyId;
    @Column(name = "positionId")
    private Integer positionId;
    @Column(name = "age")
    private Integer age;
    @Column(name = "phone")
    private String phone;
    @Column(name = "balance")
    private Integer balance;
    @Column(name = "password")
    private String password;
    @Column(name = "role")
    private String role;
    @Column(name = "gender")
    private String gender;

    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "wishes", joinColumns = {
            @JoinColumn(name = "worker_id", referencedColumnName = "id")
    }, inverseJoinColumns = {
            @JoinColumn(name = "good_id", referencedColumnName = "id")
    })
    List<Good> goods;

    @OneToMany(mappedBy = "worker", cascade = CascadeType.ALL, orphanRemoval = true)
    List<Task> tasks = new ArrayList<>();
}

File: Front/neo-store-app/src/assets/scripts/models/worker.ts
export class Worker {
    id: number;
    name: string;
    surname: string;
    secondName: string;
    login: string;
    post: number;
    position: number;
    age: number;
    phone: string;
    balance: number;
    password: string;
    role: string;
    gender: string;
    imageURL : string;
  
    constructor(
      id: number,
      name: string,
      surname: string,
      secondName: string,
      login: string,
      post: number,
      position: number,
      age: number,
      phone: string,
      balance: number,
      password: string,
      role: string,
      gender: string,
      imageURL : string
    ) {
      this.id = id;
      this.name = name;
      this.surname = surname;
      this.secondName = secondName;
      this.login = login;
      this.post = post;
      this.position = position;
      this.age = age;
      this.phone = phone;
      this.balance = balance;
      this.password = password;
      this.role = role;
      this.gender = gender;
      this.imageURL = imageURL
    }
  }